compiler/rustc_expand/src/build.rs

```diff
@@ -520,8 +521,14 @@ impl<'a> ExtCtxt<'a> {
         self.arm(span, self.pat_wild(span), self.expr_unreachable(span))
     }
 
-    pub fn expr_match(&self, span: Span, arg: P<ast::Expr>, arms: ThinVec<ast::Arm>) -> P<Expr> {
-        self.expr(span, ast::ExprKind::Match(arg, arms))
+    pub fn expr_match(
+        &self,
+        span: Span,
+        arg: P<ast::Expr>,
+        arms: ThinVec<ast::Arm>,
+        kind: MatchKind,
+    ) -> P<Expr> {
+        self.expr(span, ast::ExprKind::Match(arg, arms, kind))
```

Comment:
```suggestion
        self.expr(span, ast::ExprKind::Match(arg, arms, MatchKind::Prefix))
```
It's probably enough to hardcode the kind here, builtin macros should never generate postfix matches.